crmprtd/insert.py

```diff
@@ -54,6 +54,22 @@ def max_power_of_two(num):
     return 2 ** floor(mathlog(num, 2))
 
 
+def get_network_name(sesh, obs):
+    Obs_var = sesh.query(Variable).filter_by(id=obs.vars_id).first()
+    return Obs_var.network.name
+
+
+def obs_by_network(observations, sesh):
+    observations.sort(key=lambda obs: get_network_name(sesh, obs))
```

Comment:
Good catch that the input to `groupby` needs to be sorted. I had missed that fact.

I'd recommend using the `sorted()` builtin function here for a few reasons:
1. It's often considered better practice to *not* modify input data in place, if possible.
2. `sorted()` is more flexible in that it can take an *iterable* and it doesn't rely on the input data being a list or dict or a specific object that has a `.sort` attribute.
3. The `sorted()` builtin has received a lot of developer attention from the core devs w.r.t. performance. There's a higher chance that it might do some optimizing behind the scenes that you don't need to worry about.